src/cargo/ops/cargo_test.rs

```diff
@@ -32,14 +32,15 @@ pub fn run_tests(manifest_path: &Path,
             Some(path) => path,
             None => exe.clone(),
         };
-        let cmd = try!(compile.process(exe, &compile.package)).args(test_args);
+        let cmd = try!(compile.process(CommandType::Custom(exe.to_c_str()), &compile.package))
+                  .args(test_args);
         try!(options.compile_opts.shell.concise(|shell| {
             shell.status("Running", to_display.display().to_string())
         }));
         try!(options.compile_opts.shell.verbose(|shell| {
             shell.status("Running", cmd.to_string())
         }));
-        match cmd.exec() {
+        match CommandsExecution::exec(&mut DefaultCommands, cmd) {
```

Comment:
Doesn't work because of https://github.com/rust-lang/rust/issues/18241 https://github.com/rust-lang/rust/issues/16264